File: Question_03.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let inputCount = 0;
const arrays = [];

// Print a prompt to guide the user where to input the arrays
console.log("Enter the first array of elements separated by commas:");
// Listen for input on the first line
rl.on('line', (line) => {
  arrays.push(line.split(','));
  inputCount++;

  if (inputCount === 2) {
    rl.close();
  } else {
    // Print a prompt to guide the user where to input the second array
    console.log("Enter the second array of elements separated by commas:");
  }
});

// After all input is received, process and display the non-repeating intersection
rl.on('close', () => {
  const firstArray = arrays[0];
  const secondArray = arrays[1];
  const intersection = [];

  let i = 0;
  let j = 0;

  while (i < firstArray.length && j < secondArray.length) {
    if (firstArray[i] === secondArray[j]) {
      if (intersection.length === 0 || intersection[intersection.length - 1] !== firstArray[i]) {
        intersection.push(firstArray[i]);
      }
      i++;
      j++;
    } else if (firstArray[i] < secondArray[j]) {
      i++;
    } else {
      j++;
    }
  }

  console.log("Repeating Section each of the Arrays:");
  console.log(intersection.join(','));
});

File: Trial_02/Question_03.py
from typing import List


def find_intersection(arr1: List[int], arr2: List[int]) -> List[int]:
    """
    Find the sorted, non-repeating intersection of two arrays.

    Args:
        arr1 (List[int]): The first array.
        arr2 (List[int]): The second array.

    Returns:
        List[int]: A list containing the sorted, non-repeating intersection elements.
    """
    intersection = []
    i, j = 0, 0

    while i < len(arr1) and j < len(arr2):
        if arr1[i] == arr2[j]:
            if not intersection or arr1[i] != intersection[-1]:
                intersection.append(arr1[i])
            i += 1
            j += 1
        elif arr1[i] < arr2[j]:
            i += 1
        else:
            j += 1

    return intersection


def main():
    try:
        arr1 = list(map(int, input("Enter elements for the first array (comma-separated): ").split(",")))
        arr2 = list(map(int, input("Enter elements for the second array (comma-separated): ").split(",")))

        result = find_intersection(arr1, arr2)

        if result:
            print("Intersection:", ",".join(map(str, result)))
        else:
            print("No intersection found.")
    except ValueError:
        print("Invalid input. Please enter integers separated by commas.")


if __name__ == "__main__":
    main()
